pyzpl/__init__.py

```diff
@@ -180,6 +207,163 @@ def dumps(tree, *args, **kwargs):
     return "\n".join(lines) + "\n"
 
 
+class ZPLnode(object):
+    """ZPLnode(name=root, value=None)
+
+    This class represents a single node in a ZPL hierarchy. Per the ZPL specification:
+
+        ZPL is designed to represent a property set, where each property has a name
+        and a value. Properties are hierarchical, i.e. properties can contain other
+        properties.
+
+    Also, preserving the stream-like properties of a ZPL sequence, the order in which nodes
+    appear is preserved
+    """
+
+    def __init__(self, **kw):
+        self._name = kw.get("name", "root")
+        self._value = kw.get("value", None)
+        self._parent = kw.get("parent", None)
+        if (self._parent):
+            self._level = self._parent._level + 1
+            self._parent._children.append(self)
+        else:
+            self._level = 0
+        self._children = []
+
+    def get(self, path, filter=(None,), name_sep=':'):
```

Comment:
Good point. I will change that

Reply:
> Would it make sense to always return only the value from the `ZPLnode.get` method?

No, because the result of `get()` can legitimately be a sub-tree. When using this interface, you are aware of the node structure and care about both the value and its children.